c++/src/BlockBuffer.hh

```diff
@@ -106,18 +106,39 @@ namespace orc {
     }
 
     void resize(uint64_t size);
+
+    uint64_t getBlockSize(uint64_t blockId) const;
+
     /**
      * Requests the BlockBuffer to contain at least newCapacity bytes.
      * Reallocation happens if there is need of more space.
      * @param newCapacity new capacity of BlockBuffer
      */
     void reserve(uint64_t newCapacity);
+
     /**
      * Write the BlockBuffer content into OutputStream
      * @param output the output stream to write to
      * @param metrics the metrics of the writer
      */
     void writeTo(OutputStream* output, WriterMetrics* metrics);
+
+    /**
+     * Get the block data by block index
+     * @param blockIndex the index of blocks
+     * @param data the pointer to the block data
+     * @param size the size of the block data
+     * @return true if the block data is successfully retrieved
+     */
+    bool getBlockData(uint64_t blockIndex, void** data, uint64_t* size) const;  // add ut
+
+    /**
+     * Get next availiable memory block
+     * @param data the pointer to the block data
+     * @param size the size of the block data
+     * @return true if the block data is successfully retrieved
+     */
+    bool requestBuffer(void** buffer, int* size);  // add ut
```

Comment:
ditto

Reply:
Isn't `Block getNextBlock();` at line 91 doing the same thing?

Reply:
never mind, I'll delete it.